frontend/comment/comment.component.js

```diff
@@ -0,0 +1,190 @@
+"use strict";
+
+(function () {
+    angular.module('app')
+        .controller("CommentController", CommentController)
+        .component('comment', {
+            templateUrl: 'app/comment/comment.html',
+            controller: "CommentController",
+            controllerAs: 'commentCtrl',
+            bindings: {
+                user: '=',
+                post: '=',
+                comment: '=',
+                reply: '='
+            }
+        });
+
+    function CommentController(CommentService, MessageService, ProfileService, $state, AuthService) {
+        const commentCtrl = this;
+
+        commentCtrl.user = AuthService.getCurrentUser();
+
+        // Model to store data of a new reply on a comment
+        commentCtrl.newReply = null;
+
+        // Controll the disablement of actions
+        commentCtrl.saving = false;
+
+        commentCtrl.$onInit = function () {
+            commentCtrl.setReplyId();
+            commentCtrl.setCurrentComment();
+        };
+        
+        commentCtrl.setReplyId = function () {
+            commentCtrl.replyId = commentCtrl.reply ? commentCtrl.reply.id : null;
+        }
+
+        commentCtrl.setCurrentComment = function () {
+            commentCtrl.currentComment = commentCtrl.reply ? commentCtrl.reply : commentCtrl.comment;
+        };
+
+        commentCtrl.like = function () {
+            commentCtrl.saving = true;
+            CommentService.like(commentCtrl.post.key, commentCtrl.comment.id, commentCtrl.replyId)
```

Comment:
try to use .finally to set the saving to false. since you are doing it on both success and error situation